docs/products/opensearch/howto/get-started-with-nodejs.rst

```diff
@@ -0,0 +1,528 @@
+            body
+        }, logTitles)
+    };
+
+::
+
+    run-func index.js matchSearch title 'Tomato-garlic soup with dill'
+
+These are results which I got from the dataset:
+
+.. code-block:: text
+
+    [
+      'Mussel Soup with Avocado, Tomato, and Dill ',
+      'Garlic Soup ',
+      'Cucumber-Dill Soup with Scallions ',
+      'Cold Tomato-Thyme Soup with Grilled Garlic Croutons ',
+      'Chilled Tomato, Roasted Garlic, and Basil Soup ',
+      'Fast Favorite Garlic Dill Pickles ',
+      'Fast Favorite Garlic Dill Pickles ',
+      'Asparagus and Dill Avgolemono Soup ',
+      'Garlic Tomato Sauce ',
+      'Creamy Tomato Soup '
+    ]
+
+OpenSearch engine returned 10 most relevant results. Why 10? Because it is a default value. It can be increased by setting size property to a higher number.
+
+Next, lets use some special operators in a search query with *a query string*. Let's also increase the number of returned results to 100 to demonstrate how we can get more than 10 values.
+
+.. code-block:: javascript
+
+    /**
+     * Using special operators within a query string and a size parameter.
+     */
+    module.exports.querySearch = (query, size) => {
+        console.log(`Searching for ${query} and returning maximum ${size} results`);
+        const body = {
+            'query': {
+                'query_string': {
+                    "fields" : ["ingredients"],
+                    query
+                }
+            }
+        }
+        client.search({
+            index: indexName,
+            body,
+            size
+        }, logTitles)
+    };
+
+The full list of operators and their shortcuts can be found `in the documentation <https://opensearch.org/docs/opensearch/query-dsl/full-text/#simple-query-string>`_.
+
+To find recipes with tomato, salmon or tuna and no onion run the next query:
+
+::
+
+    run-func index.js querySearch "(salmon|tuna) +tomato -onion" 100
+
+
+Another useful feature of free-text query is defining how far search words can be from each other to still be considered a match. This parameter is called ``slop``.
+
+.. code-block:: javascript
+
+    /**
+     * Specifying a slop - a distance between search words.
+     */
+    module.exports.slopSearch = (field, query, slop) => {
+        console.log(`Searching for ${query} within distance of ${slop} in the field ${field}`);
+        const body = {
+            'query': {
+                'match_phrase': {
+                    [field]: {
+                        query,
+                        slop
+                    }
+                }
+            }
+        }
+        client.search({
+            index: indexName,
+            body
+        }, logTitles)
+    };
+
+
+Let's use this method to find recipes for pizza with pineapple. I've learned from my Italian colleagues that it is an illegal combination! Let's see if we have any recipes where words pizza and pineapple are located within the distance of maximum 10 words.
```

Comment:
Consider rewording away from "Let's" to explain the feature, and tell a joke about pineapple on pizza.